lib/sidekiq/batch.rb

```diff
@@ -12,16 +12,15 @@ class NoBlockGivenError < StandardError; end
 
     BID_EXPIRE_TTL = 108_000
 
-    attr_reader :bid, :description, :callback_queue
+    attr_reader :bid, :description, :callback_queue, :created_at
 
     def initialize(existing_bid = nil)
       @bid = existing_bid || SecureRandom.urlsafe_base64(10)
-      Sidekiq.redis do |r|
-        r.multi do
-          r.hset("BID-#{bid}", 'created_at', Time.now)
-          r.expire("BID-#{bid}", BID_EXPIRE_TTL)
-        end
-      end
+      @existing = !(!existing_bid || existing_bid.empty?)  # Basically existing_bid.present?
+      @initialized = false
+      @created_at = Time.now.utc.to_f
+      @bidkey = "BID-" + @bid.to_s
+      @ready_to_queue = []
```

Comment:
Why don't we persist it in redis? The reason I was putting that in redis directly in initializer is that we can query if the batch job is there even if it was not started.